src/management-system-v2/lib/data/users.tsx

```diff
@@ -0,0 +1,74 @@
+'use server';
+
+import { getCurrentUser } from '@/components/auth';
+import {
+  deleteUser as _deleteUser,
+  updateUser as _updateUser,
+  usersMetaObject,
+  UserHasToDeleteOrganizationsError,
+} from './legacy/iam/users';
+import { userError } from '../user-error';
+import { AuthenticatedUserData, AuthenticatedUserDataSchema } from './user-schema';
+import { ReactNode } from 'react';
+import { getEnvironmentById } from './legacy/iam/environments';
+import { OrganizationEnvironment } from './environment-schema';
+import Link from 'next/link';
+
+export async function deleteUser() {
+  const { userId } = await getCurrentUser();
+
+  try {
+    _deleteUser(userId);
+  } catch (e) {
+    let message: ReactNode = 'Error deleting user';
+
+    if (e instanceof UserHasToDeleteOrganizationsError) {
+      const conflictingOrgsNames = e.conflictingOrgs.map(
+        (orgId) => (getEnvironmentById(orgId) as OrganizationEnvironment).name,
+      );
+
+      message = (
+        <>
+          <p>
+            You're part of organizations where you are the only admin, in order to delete your
+            accounts you first have to either add another admin or delete these organizations.
```

Comment:
I don't know how consistent we are with the naming, but does the user interact with "organization" much? Most of the time we just call them spaces now, at least user-facing?

Reply:
I mean, there are some places where we use the term organization (creation and deletion), but I'm also not sure if this is the best solution. The main problem is that a user should understand that there is a difference between a personal space and an organization space.